torch/nn/quantized/modules/linear.py

```diff
@@ -229,7 +233,14 @@ def from_float(cls, mod):
         act_scale, act_zp = activation_observer.calculate_qparams()
         assert weight_observer.dtype == torch.qint8, 'Weight observer must have dtype torch.qint8'
         wt_scale, wt_zp = weight_observer.calculate_qparams()
-        qweight = torch.quantize_linear(mod.weight.float(), float(wt_scale), int(wt_zp), torch.qint8)
+        if weight_observer.qscheme in set([torch.per_tensor_symmetric, torch.per_tensor_affine]):
```

Comment:
also this block seems to be repeated as well, could you write another function that wraps around this?
e.g.
```
def quantize_weight(mod, weight_observer):
      wt_scale, wt_zp = weight_observer.calculate_qparams()
      if weight_observer.qscheme...
      return qweight
```